BlobStorage/Advanced.cs

```diff
@@ -1487,32 +1228,37 @@ private static async Task CreateNestedBlockBlobsAsync(CloudBlobContainer contain
         /// <summary>
         /// Creates a new blob and takes a snapshot of the blob.
         /// </summary>
-        /// <param name="container">A CloudBlobContainer object.</param>
+        /// <param name="container">A BlobContainerClient object.</param>
         /// <returns>A Task object.</returns>
-        private static async Task CreateBlockBlobSnapshotAsync(CloudBlobContainer container)
+        private static async Task CreateBlockBlobSnapshotAsync(BlobContainerClient container)
         {
             // Create a new block blob in the container.
-            CloudBlockBlob baseBlob = container.GetBlockBlobReference("sample-base-blob.txt");
-
-            // Add blob metadata.
-            baseBlob.Metadata.Add("ApproxBlobCreatedDate", DateTime.UtcNow.ToString());
+            BlockBlobClient baseBlob = container.GetBlockBlobClient("sample-base-blob.txt");
 
             try
             {
-                // Upload the blob to create it, with its metadata.
-                await baseBlob.UploadTextAsync(string.Format("Base blob: {0}", baseBlob.Uri.ToString()));
+                using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(string.Format("Base blob: {0}", baseBlob.Uri.ToString()))))
+                {
+                    // Upload the blob to create it, with its metadata.
+                    await baseBlob.UploadAsync(stream);
+                }
+                
+                // Add blob metadata.
+                Dictionary<string, string> metadata = new Dictionary<string, string>();
+                metadata.Add("ApproxBlobCreatedDate", DateTime.UtcNow.ToString());
+                await baseBlob.SetMetadataAsync(metadata);
```

Comment:
Metadata on upload. Also maybe you want to use BinaryData instead of MemoryStream when uploading some simple text.